### PR TITLE
spec: improve python packaging according to new guidelines

### PR DESCRIPTION
**Pull request by ignatenkobrain**
* Subpackage with python2 files now prefixed with python2- instead of python-
* Added %python_provide macro for each subpackage to correctly provide python/python2
  versions
* Fix BuildRequires/Requires to require python2 packages in f23+ instead of just python
* Removed Group tags because it's not used anywhere
* Remove hacks around copying sources for building python3 version:
  There is no point to do this, because CMake provides ability to easy do builds
  out-of-tree, so let's just have 2 build directories for py2/py3.
* Use %license macro for license
* Use %make_build and %make_install macros
* Use %python2_sitelib for python2 package instead of %python_sitelib

Signed-off-by: Igor Gnatenko <ignatenko@redhat.com>

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
Looks good to me. It's closer in-line to my [dnf-plugins-core spec for Mageia](http://svnweb.mageia.org/packages/cauldron/dnf-plugins-core/current/SPECS/dnf-plugins-core.spec?view=markup). There might be some improvements there that you might want to "steal" for Fedora. 😀

Overall, :+1:

**Comment by jsilhan** [comment]
This should go to rawhide only to avoid unintentional breaks in updates of plugins.

**Comment by mluscon** [comment]
retest please

**Comment by ignatenkobrain** [comment]
retest please

**Comment by ignatenkobrain** [comment]
retest this please

**Comment by ignatenkobrain** [comment]
@mluscon now it works with fixed tito ;)

File: dnf-plugins-core.spec [hunk 1]
```diff
@@ -2,130 +2,139 @@
...
+Version:        0.1.20
+Release:        1%{?dist}
+Summary:        Core Plugins for DNF
+License:        GPLv2+
+URL:            https://github.com/rpm-software-management/dnf-plugins-core
+Source0:        %{url}/archive/%{name}-%{version}/%{name}-%{version}.tar.gz
+BuildArch:      noarch
+BuildRequires:  cmake
+BuildRequires:  gettext
+Requires:       python-%{name} = %{version}-%{release}
```
Comment (mluscon) [inline dnf-plugins-core.spec]: Shouldn't this be python3-%{name}?

**Pull request merged**